### ISSUE TITLE
[Bug]: Call to Hugging Face w/ `n>1` returns only one output

### ISSUE THREAD

**Issue by neubig** [BerriAI/litellm#415]
### What happened?

When making a call to Hugging Face with `n` greater than 1, it still returns only one output. For example:

```python
from text_generation import Client
from litellm import completion
import json

prediction = completion(model='huggingface/chavinlo/alpaca-native', messages=[{'role': 'user', 'content': "Rewrite in the style of Shakespeare: Help I don't know what is wrong with my code!\n"}], temperature=1, max_tokens=15, top_p=1.0, n=3, api_base='http://[add_server_id]')

print(json.dumps(prediction, indent=2))
```

Results in:
```json
{
  "object": "chat.completion",
  "choices": [
    {
      "finish_reason": "length",
      "index": 0,
      "message": {
        "content": "\nAnswer: You have a syntax error in your code. You need to",
        "role": "assistant",
        "logprobs": -7.515762396
      }
    }
  ],
  "id": "chatcmpl-fb4c30ed-46e7-4279-993c-70de945080ae",
  "created": 1695090529.6827016,
  "model": "chavinlo/alpaca-native",
  "usage": {
    "prompt_tokens": 20,
    "completion_tokens": 15,
    "total_tokens": 35
  }
}
```

The expected result is that this will return three results.

Looking at the TGI client code, it seems that it is necessary to set the `best_of` parameter:
https://github.com/huggingface/text-generation-inference/blob/eeaa22ab044a5df8a570f08d2c4d971e51829b1f/clients/python/text_generation/client.py#L66

And then grab the `best_of_sequences`: https://github.com/huggingface/text-generation-inference/blob/eeaa22ab044a5df8a570f08d2c4d971e51829b1f/clients/python/text_generation/types.py#L207

**Comment by ghost** [comment]
@neubig thanks for flagging this. I'll look into this and push a fix out shortly.

**Comment by ghost** [comment]
@neubig when i call a hf TGI endpoint, even if i set best_of == 2, i just get 1 returned sequence. 

Are you seeing similar behaviour? 
<img width="1298" alt="Screenshot 2023-09-20 at 2 42 19 PM" src="https://github.com/BerriAI/litellm/assets/17561003/333c451b-f495-4df3-b0d0-bdd76f0a72f9">


Pushing logic to return all best of sequences: c63db48652324e613b3271f95a98595fdb70bce0

**Comment by neubig** [comment]
I haven't tried it myself. but is the length of `best_of_sequences` equal to 1 or 2? I can't tell from the screenshot that you posted.

**Comment by ghost** [comment]
1 
<img width="1022" alt="Screenshot 2023-09-20 at 2 49 50 PM" src="https://github.com/BerriAI/litellm/assets/17561003/1152b8a2-0afe-4260-a04e-4329de6f41ba">

**Comment by ghost** [comment]
Available for testing in v`0.1.716` 

@neubig

**Comment by ghost** [comment]
Please let me know if this works for you.

**Comment by neubig** [comment]
I can confirm with the newest version that I am still getting 1. Maybe this needs to be filed with Hugging Face TGI?

**Comment by ghost** [comment]
I think so. Do you see the same behaviour pattern as i do when you make a direct http call?

**Comment by neubig** [comment]
What program are you using to do that?

**Comment by neubig** [comment]
Ah, @krrishdholakia , I found the issue! You need to use *both* `generated_text` and all of the hypotheses in `best_of_sequences`. So the expected length of `best_of_sequences` is `n-1`.

**Comment by ghost** [comment]
Ah.. that's a bit counterintuitive. Thanks for the catch @neubig

**Comment by ghost** [comment]
fixed - a8711dc5c28f1e90f250e8753d4296c00ebd6d43

**Comment by ghost** [comment]
will close this issue once fix is published on pypi

**Comment by ghost** [comment]
Published in `0.1.720`. @neubig can you please test and let me know if this worked for you.

**Comment by neubig** [comment]
Thanks! It seemed to work for me:

```json
{
  "object": "chat.completion",
  "choices": [
    {
      "finish_reason": "length",
      "index": 0,
      "message": {
        "content": "\nAnswer: You are missing a quotation mark at the end of the",
        "role": "assistant",
        "logprobs": -7.193928237627
      }
    },
    {
      "finish_reason": "length",
      "index": 0,
      "message": {
        "content": "Same as above but instead of trying to rebuild the whole thing,",
        "role": "assistant",
        "logprobs": -37.1471825116
      }
    }
  ],
  "id": "chatcmpl-6aad4034-8fc0-4725-b521-4c64d370f489",
  "created": 1695289500.746878,
  "response_ms": 492.444,
  "model": "chavinlo/alpaca-native",
  "usage": {
    "prompt_tokens": 20,
    "completion_tokens": 14,
    "total_tokens": 34
  }
}
```

One minor thing is that the second index should probably be 1, not 0.

**Comment by ghost** [comment]
reopening issue to track the index problem. I'll close it once i've pushed the fix. thanks for the callout @neubig

**Comment by ghost** [comment]
fix pushed - 9ec8ea89bbf554548a4984d57c9b8a3d556ea558

will close issue once it's in live on pypi

**Comment by ghost** [comment]
Live in `0.1.721` - @neubig let me know if you have any issues.
